processing/src/main/java/org/apache/druid/query/GroupByMergedQueryRunner.java

```diff
@@ -114,10 +113,10 @@ public Void call()
                           {
                             try {
                               if (bySegment) {
-                                input.run(threadSafeQueryPlus, responseContext)
+                                input.run(queryPlus.withoutQueryMetrics(), responseContext)
```

Comment:
It needs to be explained now why `queryPlus.withoutQueryMetrics()` is passed down. Maybe involves some research why `queryPlus.withoutThreadUnsafeState()` used to be called in the first place.